rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -0,0 +1,211 @@
+        hourly_schedule.get("hourly_values")
+        if hourly_schedule is not None
+        else [1] * len(inlet_temperature_hourly_values)
+    )
+
+    swh_use_value = swh_use.get("use", 0.0)
+    energy_required_by_space = {}
+    volume = ZERO.VOLUME
+    for space in spaces:
+        space_id = space["id"]
+        if use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_PERSON:
+            power = swh_use_value * ureg("W") * space.get("number_of_occupants", 0)
+            energy_required = (
+                power
+                * sum(hourly_values)
+                * ureg("hr")
+                * (1 - drain_heat_recovery_efficiency)
+            )
+            energy_required_by_space[space_id] = energy_required
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_AREA:
+            power = swh_use_value * ureg("W/m2") * space.get("floor_area", ZERO.AREA)
+            energy_required = (
+                power
+                * sum(hourly_values)
+                * ureg("hr")
+                * (1 - drain_heat_recovery_efficiency)
+            )
+            energy_required_by_space[space_id] = energy_required
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER:
+            energy_required_by_space[space_id] = (
+                swh_use_value
+                * ureg("W")
+                * sum(hourly_values)
+                * ureg("hr")
+                * (1 - drain_heat_recovery_efficiency)
+            )
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.VOLUME_PER_PERSON:
+            volume += swh_use_value * ureg("L") * space.get("number_of_occupants", 0)
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.VOLUME_PER_AREA:
+            volume += swh_use_value * ureg("L/m2") * space.get("floor_area", ZERO.AREA)
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.VOLUME:
+            volume += swh_use_value * ureg("L")
+
+        else:
+            energy_required_by_space[space_id] = None
+
+        # If unit is volume based
+        if space_id not in energy_required_by_space:
+            energy_required_by_space[space_id] = sum(
+                [
+                    (volume * hourly_value * (1 - drain_heat_recovery_efficiency))
+                    * WATER_DENSITY
+                    * WATER_SPECIFIC_HEAT
+                    * (
+                        supply_temperature
+                        - inlet_temperature_hourly_values[index] * ureg("degC")
+                    )
+                    for index, hourly_value in enumerate(hourly_values)
+                ],
+                ZERO.ENERGY,
+            )
+
+    if not spaces:  # Empty list: falsey
+        if use_units == SERVICE_WATER_HEATING_USE_UNIT.OTHER:
+            energy_required_by_space["NO_SPACES_ASSIGNED"] = None
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER:
+            energy_required_by_space["NO_SPACES_ASSIGNED"] = (
+                swh_use_value * ureg("W") * ureg("hr")
```

Comment:
Why not give an energy unit here?

Reply:
Good point! Also, I think RDS is incorrect as  swh_use_value is in a unit of "W", we need to multiply by the number of hours in a year.

Reply:
Sounds good. You can update the RDS to correct the error.